src/test/java/integration/ShipmentControllerIT.java

```diff
@@ -152,25 +166,98 @@ public void createShipmentWithoutGroup() throws Exception {
                 extract().
                         path("uuid");
 
+        
         // check created data
         Shipment createdShipment = shipmentService.getEntityByUuid(UUID.fromString(newShipmentUuid),
-                sender.getCounterparty().getUser());
+                counterparty.getUser());
+        // adjust jsonObject with actual data, modified in server
+        JSONObject createdRecipient = (JSONObject) jsonObject.get("recipient");
+        createdRecipient.put("uuid", createdShipment.getRecipient().getUuid().toString());
+        createdRecipient.put("name", createdShipment.getRecipient().getName());
+        createdRecipient.put("firstName", createdShipment.getRecipient().getFirstName());
+        createdRecipient.put("middleName", createdShipment.getRecipient().getMiddleName());
+        createdRecipient.put("lastName", createdShipment.getRecipient().getLastName());
+        JSONObject createdSender = (JSONObject) jsonObject.get("sender");
+        createdSender.put("uuid", createdShipment.getSender().getUuid().toString());
+        createdSender.put("name", createdShipment.getSender().getName());
+        createdSender.put("firstName", createdShipment.getSender().getFirstName());
+        createdSender.put("middleName", createdShipment.getSender().getMiddleName());
+        createdSender.put("lastName", createdShipment.getSender().getLastName());
+        
         ObjectMapper mapper = new ObjectMapper();
         String actualJson = mapper.writeValueAsString(shipmentMapper.toDto(createdShipment));
-
+        
+        String expectedJson = jsonObject.toJSONString();
         JSONAssert.assertEquals(expectedJson, actualJson, false);
 
         // delete
         testHelper.deleteShipment(createdShipment);
     }
     
+    
+    @Test
+    @SuppressWarnings("unchecked")
+    public void createShipmentWithoutGroup_savedSender_unsavedRecipient() throws Exception {
+        Counterparty counterparty = testHelper.createCounterparty();
+        Client presavedSender = testHelper.createClient(counterparty);
+        Address recipientAddress = testHelper.createAddressOtherRegion();
+        
+        JSONObject jsonObject = testHelper.getJsonObjectFromFile("json/shipment.json");
+        jsonObject.put("sender", testHelper.toJsonClientUuidObject(presavedSender));
+        JSONObject jsonRecipient = (JSONObject) jsonObject.get("recipient");
+        jsonRecipient.put("counterpartyUuid", counterparty.getUuid().toString());
+        jsonRecipient.put("addressId", recipientAddress.getId());
+        jsonRecipient.put("uuid", null);
```

Comment:
Plz, try to use in one of your method
    jsonRecipient.remove("uuid");
instead of
    jsonRecipient.put("uuid", null);
to test this behavior as well.